runtime/src/main/scala/akka/grpc/internal/AkkaHttpClientUtils.scala

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.grpc.internal
+
+import java.net.InetSocketAddress
+import java.security.SecureRandom
+import java.util.concurrent.CompletionStage
+import scala.concurrent.duration._
+import akka.{ Done, NotUsed }
+import akka.actor.ClassicActorSystemProvider
+import akka.annotation.InternalApi
+import akka.event.LoggingAdapter
+import akka.grpc.GrpcProtocol.GrpcProtocolReader
+import akka.grpc.{ GrpcClientSettings, GrpcResponseMetadata, GrpcSingleResponse, ProtobufSerializer }
+import akka.http.scaladsl.model.HttpEntity.{ Chunk, Chunked, LastChunk, Strict }
+import akka.http.scaladsl.{ ClientTransport, ConnectionContext, Http }
+import akka.http.scaladsl.model.{ AttributeKey, HttpHeader, HttpRequest, HttpResponse, RequestResponseAssociation, Uri }
+import akka.http.scaladsl.settings.ClientConnectionSettings
+import akka.stream.{ Materializer, OverflowStrategy }
+import akka.stream.scaladsl.{ Keep, Sink, Source }
+import akka.util.ByteString
+import io.grpc.{ CallOptions, MethodDescriptor, Status, StatusRuntimeException }
+
+import javax.net.ssl.{ KeyManager, SSLContext, TrustManager }
+import scala.collection.immutable
+import scala.compat.java8.FutureConverters.FutureOps
+import scala.concurrent.{ Await, ExecutionContext, Future, Promise }
+import scala.util.{ Failure, Success }
+import akka.http.scaladsl.model.StatusCodes
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+object AkkaHttpClientUtils {
+
+  /**
+   * INTERNAL API
+   */
+  @InternalApi
+  def createChannel(settings: GrpcClientSettings, log: LoggingAdapter)(
+      implicit sys: ClassicActorSystemProvider): InternalChannel = {
+    implicit val ec = sys.classicSystem.dispatcher
+
+    log.debug("Creating gRPC client channel")
+
+    // https://github.com/grpc/grpc/blob/master/doc/compression.md
+    // since a client can't assume what algorithms a server supports, we
+    // must default to no compression.
+    // Configuring a different default could be a future feature.
+    // Configuring compression per call could be a future power API feature.
+    implicit val writer = GrpcProtocolNative.newWriter(Identity)
+
+    val target = {
+      // TODO FIXME discovery, loadbalancing etc
+      // https://github.com/akka/akka-grpc/issues/1196
+      // https://github.com/akka/akka-grpc/issues/1197
+      val resolved = settings.serviceDiscovery.lookup(settings.serviceName, 10.seconds)
+      Await.result(resolved, 10.seconds).addresses.head
```

Comment:
Ultimately not, of course - but this should be replaced when we switch to a loadbalancing API, anyway, so that's something to solve at that point I think